src/Sylius/Component/Core/Factory/CustomerFactory.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Factory;
+
+use Sylius\Component\Core\Model\AddressInterface;
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+final class CustomerFactory implements CustomerFactoryInterface
```

Comment:
It might be misleading to have `CustomerFactory` which isn't `sylius.factory.customer` but `sylius.custom_factory.customer`.

I'd propose `CustomerAfterCheckoutFactory` instead, registered at `sylius.factory.customer_after_checkout`.